roles/setup_patroni/tasks/setup_patroni.yml

```diff
@@ -0,0 +1,149 @@
+---
+- name: Set the os variable
+  ansible.builtin.set_fact:
+    os: "{{ ansible_distribution }}{{ ansible_distribution_major_version }}"
+
+- name: Check support for Operating System
+  ansible.builtin.fail:
+    msg: "Operating System = {{ os }} not supported."
+  when: os not in supported_os
+
+- name: Check supported versions for Database engine
+  ansible.builtin.fail:
+    msg: "Database Engine Version = {{ pg_version }} not supported.
+          Supported versions are {{ supported_pg_version }}"
+  when: pg_version|int not in supported_pg_version
+
+- name: Reference pg_type variables
+  ansible.builtin.include_vars: "{{ pg_type }}_{{ ansible_os_family }}.yml"
+
+- name: Reference ssl variables
+  ansible.builtin.include_vars: edb-ssl.yml
+  when: pg_ssl
+
+- name: Gather service facts
+  ansible.builtin.service_facts:
+
+- name: Open Postgres TCP port {{ pg_port }}
+  ansible.posix.firewalld:
+    port: "{{ pg_port }}/tcp"
+    permanent: true
+    state: enabled
+    immediate: true
+  when:
+    - ansible_facts.services['firewalld.service'] is defined
+    - ansible_facts.services['firewalld.service'].state == 'running'
+    - ansible_facts.services['firewalld.service'].status == 'enabled'
+  become: true
+
+- name: Open Postgres TCP port {{ patroni_rest_api_port }}
+  ansible.posix.firewalld:
+    port: "{{ patroni_rest_api_port }}/tcp"
+    permanent: true
+    state: enabled
+    immediate: true
+  when:
+    - ansible_facts.services['firewalld.service'] is defined
+    - ansible_facts.services['firewalld.service'].state == 'running'
+    - ansible_facts.services['firewalld.service'].status == 'enabled'
+  become: true
+
+- name: Gather the patroni cluster informations
+  ansible.builtin.set_fact:
+    etcd_cluster_nodes: "{{ lookup('edb_devops.edb_postgres.etcd_nodes', wantlist=True) }}"
```

Comment:
Is there a check or way to ensure that all patroni nodes are in the list of `etcd_cluster_nodes` before progressing further?

Reply:
Do all Patroni nodes require etcd? If so, how are you sure that the Patroni nodes gathered in `pg_sr_cluster_nodes` have `etcd: true` set in the inventory file and are all apart of the same etcd cluster?

Reply:
Good point, though!